### ISSUE TITLE
EntryComplete_Sources

### ISSUE THREAD

**Issue by devpew** [inkarkat/vim-CompleteHelper#1]
Hello inkarkar, you wrote a really nice plugin for vim. I spent about 2 days to find something like that. I'm talking about EntryComplete module.

Could you help me with one moment

It works fine when i use it with one file, like that
`
let g:EntryComplete_Sources = [bufnr('lines.txt'), expand('~/.vimrc'), expand("~/Nextcloud/txt/1.txt")]`

But it doesn't work if i want to use all the files in the folder. 

For example if i have a lot of *.txt files in one folder 

Could you help me with that?

**Comment by inkarkat** [comment]
You're right; if there are many (and a changing amount of) files in a folder, it is tedious to enumerate them manually. Fortunately, Vim has a `glob()` function that automatically resolves a file glob (like `*.txt`) into a List of filespecs. `g:EntryComplete_Sources` takes a List of filespecs, so these just need to be combined:

    let g:EntryComplete_Sources = ingo#compat#glob('~/path/to/*.txt', 0, 1)

This will only resolve the list of filespecs once when Vim starts.

### alternative

If you need a more dynamic lookup, you can install a Funcref to a function that does the file lookup whenever the entry completion is triggered:


    let g:EntryComplete_Sources = [function('MyEntryLookup')]
    function! MyEntryLookup()
        return ingo#compat#glob('~/path/to/*.txt', 0, 1)
    endfunction

I hope this helps. Please comment / reopen if you have further questions!

**Comment by devpew** [comment]
I tried those 2 ways, but it doesn't work

Config:

```
Plugin 'inkarkat/vim-CompleteHelper'
Plugin 'inkarkat/vim-ingo-library'
Plugin 'MarcWeber/vim-addon-mw-utils'
Plugin 'EntryComplete'

let g:EntryComplete_Sources = ingo#compat#glob('~/Nextcloud/txt/projects/*.txt', 0, 1)
```


Error on load:

```
Error detected while processing /home/dm/.vimrc:
line   21:
E117: Unknown function: ingo#compat#glob
E15: Invalid expression: ingo#compat#glob('~/Nextcloud/txt/projects/*.txt', 0, 1)
Press ENTER or type command to continue
```




Maybe I should load another mobules ?

**Comment by devpew** [comment]
I thied to remove everything from my comfig file
I tried compatible and uncompatible modes

But it doesn't work :(

It works great with let g:EntryComplete_Sources = [expand('~/.vimrc')]

But doesnt't work with ingo#compat#glob('~/Nextcloud/txt/projects/*.txt', 0, 1)

**Comment by inkarkat** [comment]
Hm, it looks like the Plugin manager only adds the paths to `'runtimepath'` later; so the ingo-library functions aren't yet available in your `.vimrc`; the same problem happens with Vim 8 _pack plugins_; that's why I've moved plugin customizations into a separate `~/.vim/plugin/00plugins.vim` file. That would help in your case, too.

However, if you're only using Vim 8.0 and later, you can simply drop the compatibility wrapper and directly use `glob()` instead of `ingo#compat#glob()`.

**Comment by inkarkat** [comment]
BTW, I've just released version 1.20 of EntryComplete, now [also on GitHub](https://github.com/inkarkat/vim-EntryComplete). You may want to update after you've hopefully resolved this problem.

**Comment by devpew** [comment]
Thank you very much! It works absolutely amazing :+1: 
I'll try to update plugin, thank you!
